server/routes/api/reviews.js

```diff
@@ -111,4 +111,34 @@ router.get("/requests/:id", authenticate, async (req, res) => {
   }
 });
 
+router.put("/:id", authenticate, async (req, res) => {
+  const requestId = req.params.id;
+  const { messageId, code, comments, authorId } = req.body;
+
+  try {
+    const userId = req.user.id;
+
+    const singleRequest = await Review.find({
+      _id: { $in: [requestId] },
+      requesterId: { $in: [userId] },
+    });
+
+    if (!singleRequest[0]) {
+      return res.sendStatus(400);
+    }
+
+    singleRequest[0].messages.map((message) => {
+      if (message["_id"] == messageId && message.authorId == authorId) {
+        message.code = code;
+        message.comments = comments;
+      }
+    });
+
+    await singleRequest[0].save();
+    return res.sendStatus(200);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
```

Comment:
This route should instead be `/api/messages/:id` (or maybe `/api/reviews/messages/:id`) since we're modifying the message and not the review itself. Get the message id from `req.params` and query the Message collection instead. Should also use `Message.findById()` and check if you get a document and if the authorId matches the authenticated user. No query result should return a status 404 and mismatching user ids should return a status 403 for forbidden.